Userland/Libraries/LibCrypto/ASN1/DER.cpp

```diff
@@ -210,187 +210,114 @@ Optional<DecodeError> Decoder::enter()
     return {};
 }
 
-Optional<DecodeError> Decoder::leave()
+ErrorOr<void> Decoder::leave()
 {
     if (m_stack.is_empty())
-        return DecodeError::NoInput;
+        return Error::from_string_literal("Leaving using an empty stack");
 
     if (m_stack.size() == 1)
-        return DecodeError::LeavingMainContext;
+        return Error::from_string_literal("Leaving the main context");
 
     m_stack.take_last();
     m_current_tag.clear();
 
     return {};
 }
 
-void pretty_print(Decoder& decoder, DeprecatedOutputStream& stream, int indent)
+ErrorOr<void> pretty_print(Decoder& decoder, DeprecatedOutputStream& stream, int indent)
 {
     while (!decoder.eof()) {
-        auto tag = decoder.peek();
-        if (tag.is_error()) {
-            dbgln("PrettyPrint error: {}", tag.error());
-            return;
-        }
+        auto tag = TRY(decoder.peek());
 
         StringBuilder builder;
         for (int i = 0; i < indent; ++i)
             builder.append(' ');
```

Comment:
A bunch of these appends could become `TRY(try_append...)` but that could definitely be a future thing.

Reply:
I opted to keep it as minimal as possible for now, but yeah, it would definitely be an improvement in the future. :^)